README.md

```diff
@@ -247,35 +295,80 @@ std::optional<lex_item> lexer(std::string_view v) noexcept {
 This support is preliminary, probably the API will be changed.
 
 ```c++
-auto input = "123,456,768"sv;
+#include <ctre.hpp>
+#include <iostream>
+#include <string_veiw>
 
-for (auto match: ctre::range<"([0-9]+),?">(input)) {
+// auto input = "123,456,768";
+extern const char *input;
+
+int main(void) {
+  auto matches = ctre::range<"([0-9]+),?">(input);
+  for (auto match : matches) {
     std::cout << std::string_view{match.get<0>()} << "\n";
+  }
+  return 0;
 }
 ```
+[link to compiler explorer](https://gcc.godbolt.org/z/s4zedb68n)
 
 ### Unicode
 
 ```c++
 #include <ctre-unicode.hpp>
 #include <iostream>
+
 // needed if you want to output to the terminal
 std::string_view cast_from_unicode(std::u8string_view input) noexcept {
-    return std::string_view(reinterpret_cast<const char *>(input.data()), input.size());
+  return std::string_view(reinterpret_cast<const char *>(input.data()), input.size());
 }
-int main()
-{
-    using namespace std::literals;
-    std::u8string_view original = u8"Tu es un génie"sv;
-
-    for (auto match : ctre::range<"\\p{Letter}+">(original))
-        std::cout << cast_from_unicode(match) << std::endl;
-    return 0;
+
+int main() {
+  using namespace std::literals;
+  std::u8string_view original = u8"Tu es un génie"sv;
+
+  for (auto match : ctre::range<"\\p{Letter}+">(original))
+    std::cout << cast_from_unicode(match) << std::endl;
+  return 0;
 }
 ```
 
 [link to compiler explorer](https://godbolt.org/z/erTshe6sz)
 
+## Integration
+You can get [ctre.hpp](https://github.com/hanickadot/compile-time-regular-expressions/blob/main/single-header/ctre.hpp) from the directory `single-header`.
+You need to add:
+
+```C++
+#include "ctre.hpp"
+
+//using namespace ctre;
+```
+This header can be regenerated with `make single-header`.
+### CMake
+If you are using cmake, you can add this directory as subdirectory and link to target `ctre`.
+
+```CMake
+cmake_minimum_required(VERSION 3.8.0)
+
+project(MyProject VERSION 1.0)
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD_REQUIRED True)
+
+FetchContent_Declare(
+        ctre
+        GIT_REPOSITORY https://github.com/hanickadot/compile-time-regular-expressions.git
+        GIT_TAG 95c63867bf0f6497825ef6cf44a7d0791bd25883 # v3.4.1
+)
+
+FetchContent_MakeAvailable(ctre)
+include_directories("${ctre_SOURCE_DIR}/single-header")
+
+# Add an executable with the above sources
+add_executable(${PROJECT_NAME} main.cpp)
+target_link_libraries(${PROJECT_NAME})
```

Comment:
```suggestion
add_executable(example main.cpp)
target_link_libraries(example PRIVATE ctre::ctre)
```

Reply:
I feel that the CMake example would be more of a template if we did
add_executable(${PROJECT_NAME} main.cpp)
target_link_libraries(${PROJECT_NAME} PRIVATE ctre::ctre)

Reply:
Using `${PROJECT_NAME}` is pointless either way.

Reply:
`${PROJECT_NAME}` makes it so there are fewer things to rewrite.

Reply:
Find and replace exists. How often does the project name change that this abstraction needs to exist? This also just obfuscates the target's name. There are only downsides to using this.

Reply:
The project name changes every time someone copies the CMake example template and puts it in their own project. I would say that anyone wanting to not use the variable `${PROJECT_NAME}` is already further along in their project where they would only need to <kbd>Ctrl</kbd>+<kbd>C</kbd> the middle portion of the template.

Reply:
If this is just for a copy-paste example, then using an explicit name makes no semantic difference, but syntactically aligns better with what one should write to begin with. Anyone sufficiently knowledgeable regarding CMake will not bother with this example, as the only interesting thing is the exported target(s) in the install interface.